cpp/include/raft/matrix/detail/linewise_op.cuh

```diff
@@ -21,11 +21,28 @@
 #include <raft/util/vectorized.cuh>
 
 #include <algorithm>
+#include <thrust/tuple.h>
 
 namespace raft {
 namespace matrix {
 namespace detail {
 
+/** This type simplifies returning arrays and passing them as arguments */
+template <typename Type, int VecElems>
+struct VecArg {
```

Comment:
This was precisely meant to replace `TxN` because of cases like `MatT`=4B and `VecT`=8B. If `VecBytes`=4B we can't use `TxN` as `VecElems` would be 0.

Reply:
I don't get it. Why wouldn't we parameterize TxN by type of the vector element, for each vector? Then if, for example, matrix elements are uint8_t, and two vector arguments are float and double, we'd be limited to `VecElems = 2`, and the size of reads 2, 8,  16 for the matrix and vector arguments correspondingly.

Reply:
I think I see what you mean. What I tried previously that didn't work was to use `Linewise<VecT, IdxT, VecBytes, BS>::Vec` but what I should have used was `TxN_t<VecT, VecElems>`. The difference here is that `VecElems` will be `VecBytes / MatT` instead of `VecBytes / VecT`.

Reply:
As discussed offline, using `TxN` with the vector type but `VecElems` computed from the matrix type is currently not possible, because you can e.g have a situation where you want a vector type of 4 doubles, which isn't possible.

@achirkin suggested supporting more than one vector per `TxN` but I think it's outside of the scope of this PR and my change doesn't have any visible performance impact (reading the vector elements in shared mem is an insignificant operation).

Reply:
@Nyrio could you add an issue about this, so that we don't forget it?

Reply:
#990